src/oedisi/tools/cli_tools.py

```diff
@@ -156,57 +156,97 @@ def validate_optional_inputs(
+        ),
+    )
 
-    service = {
-        "apiVersion": "v1",
-        "kind": "Service",
-        "metadata": {
-            "name": f"{APP_NAME}-service",
-        },
-        "spec": {
-            "type": "NodePort",
-            "selector": {"app": APP_NAME},
-            "ports": [{"port": broker_port, "nodePort": node_port}],
-        },
-    }
-
-    deployment = {
-        "apiVersion": "apps/v1",
-        "kind": "Deployment",
-        "metadata": {"name": f"{APP_NAME}-deployment", "labels": {"app": APP_NAME}},
-        "spec": {
-            "replicas": 1,
-            "selector": {"matchLabels": {"app": APP_NAME}},
-            "template": {
-                "metadata": {"labels": {"app": APP_NAME}},
-                "spec": {"containers": []},
-            },
-        },
-    }
+    service_dict = drop_null_values(service.to_dict())
+    with open(os.path.join(kube_folder, f"service.yml"), "w") as f:
+        yaml.dump(service_dict, f)
 
-    container = {}
     for component in wiring_diagram.components:
-        container["name"] = component.name.replace("_", "-")
-        container["image"] = component.image
-        container["ports"] = [{"containerPort": component.container_port}]
-        deployment["spec"]["template"]["spec"]["containers"].append(container.copy())
-
-    container["name"] = BROKER_SERVICE.replace("_", "-")
-    container["image"] = f"{DOCKER_HUB_USER}/{APP_NAME}_{BROKER_SERVICE}:latest"
-    container["ports"] = [{"containerPort": broker_port}]
-    deployment["spec"]["template"]["spec"]["containers"].append(container.copy())
-
-    with open(os.path.join(kube_folder, "deployment.yml"), "w") as f:
-        yaml.dump(deployment, f)
-    with open(os.path.join(kube_folder, "service.yml"), "w") as f:
-        yaml.dump(service, f)
 
+        fixed_container_name =  component.name.replace("_", "-")
+        my_container = client.V1Container(            
+            name = fixed_container_name,
+            image = component.image,
+            env = [
+                client.V1EnvVar(
+                    name="PORT", 
+                    value=str(component.container_port)
+                ),
+                client.V1EnvVar(
+                    name="SERVICE_NAME", 
+                    value=SERVICE_NAME,
+                )
+            ],
+            ports =  [
+                client.V1ContainerPort(
+                    container_port = component.container_port
+                )],
+            )
+       
+        pod = client.V1Pod(
+            api_version="v1",
+            kind="Pod",
+            metadata=client.V1ObjectMeta(
+                name=f"{fixed_container_name}-pod",
+                labels ={"app" : APP_NAME},
+            ),
+            spec=client.V1PodSpec(
+                containers=[my_container],
+                hostname=component.name.replace("_", "-"),
+                subdomain="oedisi-service",
+                #node_port=int(component.container_port), #maybe
```

Comment:
We probably want to remove this comment or adjust it.

Reply:
removed